docs/effects/lighting-effect.md

```diff
@@ -0,0 +1,114 @@
+# LightingEffect
+
+The Lighting Effect applies ambient, point and directional lighting to layers which support material property.
+
+```js
+import DeckGL, {GeoJsonLayer, LightingEffect} from '@deck.gl/core';
+import {AmbientLight, PointLight, DirectionalLight, PhongMaterial} from '@luma.gl/core';
+
+// create ambient light source
+const ambientLight = new AmbientLight({
+  color: [255, 255, 255],
+  intensity: 1.0
+});
+// create point light source
+const pointLight = new PointLight({
+  color: [255, 255, 255],
+  intensity: 2.0,
+  // use coordinate system as the same as view state
+  position: [-125, 50.5, 5000]
+});
+// create directional light source
+const directionalLight = new DirectionalLight({
+  color: [255, 255, 255],
+  intensity: 1.0,
+  direction: [-3, -9, -1]
+});
+
+const lightingEffect = new LightingEffect({ambientLight, pointLight, directionalLight});
+
+const material =  new PhongMaterial({
+  ambient: 0.2,
+  diffuse: 0.5,
+  shininess: 32,
+  specularColor: [255, 255, 255]
+});
+
+const INITIAL_VIEW_STATE = {
+  latitude: 49.254,
+  longitude: -123.13,
+  zoom: 11,
+  pitch: 45
+};
+
+const deckgl = new Deck({
+  canvas: 'my-deck-canvas',
+  initialViewState: INITIAL_VIEW_STATE,
+  controller: true,
+  // add lighting effect to deck
+  effects: [lightingEffect],
+  layers: [
+    new GeoJsonLayer({
+        id: 'geojson-layer',
+        // layer props
+        ...
+        // lighting only applies to extruded polygons
+        extruded: true,
+        // specify material properties per layer
+        material
+    })
+  ]
+});
+```
+
+## Constructor
+
+```js
+new LightingEffect({light0, light1, light2, ...});
+```
+
+Parameters:
+* props(Object): a collection of light sources with user defined keys.
```

Comment:
Keys can be any arbitrary name and values are instances of `Light`.
* list link to each light source class and documentation